server/routes/api.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const router = express.Router()
+const moment = require('moment')
+const Expense = require('./model/Expense')
+
+const getDate = (isExist) => (isExist ? isExist.format('LLLL') : moment().format('LLLL'))
+
+router.get('/expenses', function (req, res) {
+    let { date1 } = req.query
+    let { date2 } = req.query
+    let d1 = Date.parse(date1)
+    let d2 = Date.parse(date2)
+
+    if (date1 && date2) {
+        Expense.find({ date: { $gte: d1, $lte: d2 } })
+            .sort({ date: 1 })
+            .exec(function (err, expenses) { res.send(expenses) })
+    }
+    else if (date1 && !date2) {
+        Expense.find({ date: { $gte: d1, $lte: Date.parse(moment().format('YYYY-MM-DD')) } })
+            .sort({ date: 1 })
+            .exec(function (err, expenses) { res.send(expenses) })
+    }
+    else {
+        Expense.find({})
+            .sort({ date: -1 })
+            .exec(function (err, expenses) { res.send(expenses) })
+    }
+})
+
+router.post('/new', function (req, res) {
+    let expense = req.body
+    let e1 = new Expense({
+        name: expense.name,
+        amount: expense.amount,
+        group: expense.group,
+        date: getDate(expense.date)
+    })
```

Comment:
better write it like this:

const { name, amount, group, date } = req.body
const e1 = new Expense({ name, amount, group, date: getDate(date) })